mitodl/bootcamp-ecommerce
# Update FluidReview with total amounts paid

#### What are the relevant tickets?
Closes #140
Closes #146

#### What's this PR do?
Submits total amount paid to the appropriate FluidReview submission metadata when a user pays for a klass and the order is fulfilled via the `OrderFulfillmentView`

#### How should this be manually tested?
- Add the following env variable: `FLUIDREVIEW_AMOUNTPAID_ID=22977` (27007 for production).
- Create a `Bootcamp` with a `Klass` (`klass_key=72713`) and `Installment`. Do not check the 'Legacy' box.
- Send a `WebhookRequest` to your bootcamp instance at `../api/v0/fluidreview_webhook/` with this data:
```json
{
  "date_of_birth": "",
  "user_email": "staff@example.com",
  "amount_to_pay": "",
  "user_id": 95292895,
  "submission_id": 4621059,
  "award_id": 72713,
  "award_cost": "1000",
  "award_name": "TEST CAMP"
}
```
- Via the front-end, submit a payment of $25.00 for this klass. 
- Via the admin page, change the `Order` status to 'Fulfilled'
- From a django shell:
  ```python
  from ecommerce.models import Order
  from fluidreview.api import *
  order = Order.objects.get(id=<order_id>)
  post_payment(order)

  api = FluidReviewAPI()
  api.get('submissions/4621059/metadata/').json()
  ```
The result should include: 
```python
{'id': 22977, 'name': 'Amount Paid', 'value': '25.00'},
```
